### PR TITLE
move common documentation to a fragment

### PR DESCRIPTION
**Pull request by evgeni**
As of Ansible 2.8, we can ship custom documentation fragments together
with our modules [1], so let's do this.

(Documentation fragments were supported for upstream Ansible modules
since at least 2.3, but you couldn't add more paths to search for them
until 2.8)

[1] https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-fragments

### CODE REVIEW

**Comment by sean797** [comment]
Nice :+1: we definitely need to wait until 2.8 is released, but that's fine as this is WIP. I suspect we should probably also wait until 2.9 is released as this is backwards incompatible?

**Comment by evgeni** [comment]
This is backwards incompatible, yes. But it also "only" breaks `ansible-doc` invocations on pre-2.8 installs.

Given that we have online-accessible docs now, I'd actually merge this as soon as 2.8 hits stable (and probably Fedora :))

**Comment by evgeni** [comment]
@sean797 @mdellweg this should now pass tests and everything.

**Comment by mdellweg** [comment]
I guess, we need to bump the ansible dependency to 2.8 everywhere.

**Comment by evgeni** [comment]
> I guess, we need to bump the ansible dependency to 2.8 everywhere.

As far as I can see the modules still work fine in older Ansible versions, just `ansible-doc` is broken as it cannot find the snippet. IMHO this warrants a warning in the docs, but no strict dependency.

**Comment by mdellweg** [comment]
> As far as I can see the modules still work fine in older Ansible versions, just `ansible-doc` is broken as it cannot find the snippet. IMHO this warrants a warning in the docs, but no strict dependency.

OK in that respect, the `module_utils` (v2.3) change remains the reasoning for this dependency, i guess.

**Comment by evgeni** [comment]
I've added a line to the README explaining the situation.

If I get a tentative ACK on the change as such, I'll go ahead an modify all the modules :)

**Comment by mdellweg** [comment]
I am all in for this change.

**Comment by ekohl** [comment]
:+1: Looks like a nice deduplication.

**Comment by evgeni** [comment]
this is now ready for review.

I only changed the modules that use the Foreman(Apypie)AnsibleModule interface, as these are the only ones that get all the updated params anyways.

I also took the liberty to yank all the mentions of ansible and python in the requirements, because I found them silly :)

File: plugins/modules/katello_content_view_filter.py [hunk 1]
```diff
@@ -126,6 +106,7 @@
     description:
       - Create an include filter
     default: False
+extends_documentation_fragment: foreman
```
Comment (ekohl) [inline plugins/modules/katello_content_view_filter.py]: This is not an apypie module. Should this still be replaced? There's a few more of those below.

Comment (evgeni) [reply plugins/modules/katello_content_view_filter.py]: Yes. It's a `ForemanAnsibleModule` (so nailgun, but still with the common wrapper API), which inherits from `ForemanBaseAnsibleModule` which has the common params:

https://github.com/theforeman/foreman-ansible-modules/blob/a2aac6240a419cb87876f9c91a062ecb754c6878/plugins/module_utils/foreman_helper.py#L28-L36

**Comment by evgeni** [comment]
Ansible upstream does not support Python 2.6 anymore, we don't test on Python 2.6, so I'd say it's safe to say we're 2.7 only. This is however not publicly documented. Should it? With the "Ansible doesn't support <2.7" argument, I'd say no.

File: plugins/modules/foreman_bookmark.py [hunk 2]
```diff
@@ -36,25 +36,6 @@
   - "ansible >= 2.3"
   - "apypie"
```
Comment (mdellweg) [inline plugins/modules/foreman_bookmark.py]: It should be possible, to move the requrements to the doc fragment, i think.
I guess, they will even get merged. If specified here too.

**Comment by ekohl** [comment]
> Ansible upstream does not support Python 2.6 anymore, we don't test on Python 2.6, so I'd say it's safe to say we're 2.7 only. This is however not publicly documented. Should it? With the "Ansible doesn't support <2.7" argument, I'd say no.

We could state that we follow Ansible's Python support. That also means that when they drop 2.7, we can drop it. Similar for Python 3 equivalents.

File: plugins/modules/foreman_bookmark.py (hunk 2 above)
Comment (evgeni) [reply plugins/modules/foreman_bookmark.py]: it is, but I didn't do that for the sake of not having two fragments (nailgun, apypie) for now.

Comment (mdellweg) [reply plugins/modules/foreman_bookmark.py]: So we postpone that. OK for me.

Comment (mdellweg) [reply plugins/modules/foreman_bookmark.py]: I guess inheriting doc fragements is not really worth it on the string level. ;)

Comment (evgeni) [reply plugins/modules/foreman_bookmark.py]: Yeah, I thought about that, but it didn't feel like a worthy endeavor for a one-line thing.

**Comment by evgeni** [comment]
@ekohl @mdellweg updated for the latest modules and added a line about supported pythons. :green_apple:

File: plugins/modules/redhat_manifest.py [hunk 3]
```diff
@@ -25,10 +25,6 @@
     - Download and modify a Red Hat Satellite Subscription Manifest
 author:
     - "Sean O'Keeffe (@sean797)"
-requirements:
-    - "nailgun >= 0.28.0"
```
Comment (sean797) [inline plugins/modules/redhat_manifest.py]: :laughing:

**Comment by mdellweg** [comment]
It has been an honour to merge this.
Thank you, @evgeni !

File: plugins/modules/redhat_manifest.py (hunk 3 above)
Comment (evgeni) [reply plugins/modules/redhat_manifest.py]: Yeah, I have NO idea how that has sneaked in ;)

**Pull request merged**